src/huggingface_hub/hf_file_system.py

```diff
@@ -1069,6 +1069,7 @@ def read(self, length: int = -1):
             )
             hf_raise_for_status(self.response)
         try:
+            self.response.raw.decode_content = True
```

Comment:
can you also add a test in [test_hf_file_system.py](https://github.com/huggingface/huggingface_hub/blob/main/tests/test_hf_file_system.py) with your repro script


```py
    fs = HfFileSystem()
    with fs.open(f"datasets/math_qa/math_qa.py", "r", encoding="utf-8") as f:
        print(f.read())
```

that checks it runs without failure?

Reply:
Thanks! I think it's fine to run against production for such edge case test. It's not the only test where we do require production environment and so far it has been working correctly. As long as it's "read-only" it's perfectly ok. The load induced by the CI is neglictable compared to the Hub traffic in general (and worse case, it'll get rate limited before getting problematic)